File: frontend/src/types/job.ts
import { z } from "zod";
import { PaginatedResponse } from "./pagination";

export const JobSummarySchema = z.object({
  id: z.string().uuid(),
  title: z.string(),
  department: z.string().nullable().optional(),
  location: z.string().nullable().optional(),
  employment_type: z.string().nullable().optional(),
  status: z.string(),
  created_at: z.string().datetime(),
  updated_at: z.string().datetime(),
  candidate_count: z.number().default(0),
  shortlisted_count: z.number().default(0),
  interview_count: z.number().default(0),
  hired_count: z.number().default(0),
});

export const JobDetailsSchema = z.object({
  id: z.string().uuid(),
  title: z.string(),
  department: z.string().nullable().optional(),
  location: z.string().nullable().optional(),
  employment_type: z.string().nullable().optional(),
  hiring_manager: z.string().nullable().optional(),
  status: z.string(),
  description: z.string(),
  skills_required: z.array(z.string()).default([]),
  experience_required: z.string().nullable().optional(),
  created_at: z.string().datetime(),
  updated_at: z.string().datetime(),
});

export type JobSummaryDTO = z.infer<typeof JobSummarySchema>;
export type JobDetailsDTO = z.infer<typeof JobDetailsSchema>;

export interface JobCandidateMatchDTO {
  id: string;
  candidate_id: string;
  first_name?: string | null;
  last_name?: string | null;
  email?: string | null;
  semantic_score: number;
  skills_score: number;
  experience_score: number;
  education_score: number;
  quality_score: number;
  final_score: number;
  strengths: string[];
  gaps: string[];
  recommendations: string[];
  created_at: string;
}

export interface JobWorkflowDTO {
  id: string;
  workflow_name: string;
  workflow_version: string;
  status: string;
  current_node?: string | null;
  started_at?: string | null;
  completed_at?: string | null;
  candidate_id?: string | null;
  last_error?: string | null;
}

export interface JobAnalyticsDTO {
  total_candidates: number;
  average_match_score: number;
  candidates_by_status: Record<string, number>;
  top_skills_matched: string[];
}

export type JobListResponse = PaginatedResponse<JobSummaryDTO>;

File: backend/src/application/schemas/job.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import Optional, List

class JobSummaryDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    title: str
    department: Optional[str] = None
    location: Optional[str] = None
    employment_type: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime
    
    # Aggregated Candidate Counts
    candidate_count: int = 0
    shortlisted_count: int = 0
    interview_count: int = 0
    hired_count: int = 0

class JobDetailsDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    title: str
    department: Optional[str] = None
    location: Optional[str] = None
    employment_type: Optional[str] = None
    hiring_manager: Optional[str] = None
    status: str
    description: str
    skills_required: List[str] = []
    experience_required: Optional[str] = None
    created_at: datetime
    updated_at: datetime

class JobCandidateMatchDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    candidate_id: UUID
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    email: Optional[str] = None
    semantic_score: float
    skills_score: float
    experience_score: float
    education_score: float
    quality_score: float
    final_score: float
    strengths: List[str] = []
    gaps: List[str] = []
    recommendations: List[str] = []
    created_at: datetime

class JobWorkflowDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    workflow_name: str
    workflow_version: str
    status: str
    current_node: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    candidate_id: Optional[UUID] = None
    last_error: Optional[str] = None

class JobAnalyticsDTO(BaseModel):
    total_candidates: int = 0
    average_match_score: float = 0.0
    candidates_by_status: dict[str, int] = {}
    top_skills_matched: list[str] = []

class JobFeedbackDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    candidate_id: UUID
    candidate_name: Optional[str] = None
    decision: str
    confidence: float
    reason: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime

class JobHistoryDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    created_at: datetime
    # Could include search parameters if added to SearchSessionModel in the future

class JobDocumentDTO(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    file_path: str
    file_type: str
    original_name: str
    created_at: datetime
